### PR TITLE
ci(shared-data, components): add github action for publishing packages to npm

### PR DESCRIPTION
**Pull request by b-cooper**
# Overview

In order to support a single source of truth for applications outside of the monorepo (read:
"Protocol Library") that rely on fundamental components and geometries within the mono repo, add tag
based deploy actions that publish bundled artifacts to NPM.

# Changelog

# Review requests

# Risk assessment

### CODE REVIEW

File: .github/workflows/components-test-build-deploy.yaml [hunk 1]
```diff
@@ -137,3 +158,41 @@ jobs:
...
         run: |
           aws s3 sync ./dist s3://opentrons-components/${{ env.OT_BRANCH}} --acl public-read
+
+  publish-components:
+    name: 'publish components package to npm'
+    runs-on: 'ubuntu-latest'
+    needs: ['js-unit-test', 'determine-build-type']
+    if: needs.determine-build-type.outputs.type == 'publish'
+    env: 
+      VERSION_STRING: echo github.ref | sed 's/refs\/tags\/components@//'
```
Comment (sfoster1) [inline .github/workflows/components-test-build-deploy.yaml]: I don't think you can drop shell commands here, I think it's [expressions](https://docs.github.com/en/actions/learn-github-actions/expressions#functions) only. i'd recommend adding the tag value as an output from the `determine-build-type` job, where you can use shell

File: .github/workflows/components-test-build-deploy.yaml [hunk 2]
```diff
@@ -137,3 +158,41 @@ jobs:
...
+            ${{ github.workspace }}/.yarn-cache
+            ${{ github.workspace }}/.npm-cache
+          key: js-${{ secrets.GH_CACHE_VERSION }}-${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
+          restore-keys: |
+            js-${{ secrets.GH_CACHE_VERSION }}-${{ runner.os }}-yarn-
+      - name: 'set version number'
+        run: |
+          npm install -g json
+          json -I -f ./components/package.json -e "this.version=\"$VERSION_STRING\""
+      - run: cd ./components && npm publish --provenace --access public
```
Comment (sfoster1) [inline .github/workflows/components-test-build-deploy.yaml]:
```suggestion
      - run: cd ./components && npm publish --provenance --access public
```

File: Makefile [hunk 3]
```diff
@@ -61,7 +61,6 @@ setup-js:
 	yarn
 	$(MAKE) -C $(APP_SHELL_DIR) setup
 	$(MAKE) -C $(APP_SHELL_ODD_DIR) setup
-	$(MAKE) -C $(SHARED_DATA_DIR) setup-js
```
Comment (b-cooper) [inline Makefile:64]: this call is no longer necessary.

File: shared-data/Makefile [hunk 4]
```diff
@@ -28,17 +25,12 @@ clean: clean-js clean-py
...
 # JavaScript targets
 
-.PHONY: setup-js
-setup-js: dist-js
+.PHONY: lib-js
+lib-js: export NODE_ENV := production
+lib-js:
+	yarn webpack
 
-.PHONY: dist-js
```
Comment (b-cooper) [inline shared-data/Makefile:34]: these were purely used for bundling v1 labware defs for old versions of monorepo SPA's

File: protocol-library-kludge/src/URLDeck.tsx [hunk 5]
```diff
@@ -115,14 +106,7 @@ export class URLDeck extends React.Component<{}> {
                   >
                     {labwareDefV2 ? (
                       <LabwareRender definition={labwareDefV2} />
-                    ) : (
```
Comment (b-cooper) [inline protocol-library-kludge/src/URLDeck.tsx:118]: this whole directory will never be deployed again. It can be removed when legacy PL is unlisted

Comment (sfoster1) [reply protocol-library-kludge/src/URLDeck.tsx:118]: We eagerly await the day

**Review by sfoster1 — APPROVED** [review]
Looks good to me!

File: .github/workflows/components-test-build-deploy.yaml [hunk 6]
```diff
@@ -102,11 +102,32 @@ jobs:
...
+              echo 'type=storybook' >> $GITHUB_OUTPUT
+            elif [ "${{ format('{0}', startsWith(github.ref, 'refs/tags/components')) }}" = "true" ] ; then
+              echo "publish builds for components tags"
+              echo 'type=publish' >> $GITHUB_OUTPUT
+            else
+              echo "No build for ref ${{github.ref}} and event ${{github.event_type}}"
+              echo 'type=none' >> $GITHUB_OUTPUT
+            fi
+
   deploy-components:
```
Comment (shlokamin) [inline .github/workflows/components-test-build-deploy.yaml:126]:
```suggestion
  deploy-components-storybook:
```

**Review by shlokamin — APPROVED** [review]
🥳

**Pull request merged**